pkg/audit/stats_reporter.go

```diff
@@ -0,0 +1,124 @@
+package audit
+
+import (
+	"context"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/metrics"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	totalViolationsName  = "violations_total"
+	constraintsTotalName = "constraints_total"
+	auditLatency         = "audit_latency"
+	methodType           = "audit"
+)
+
+var (
+	violationsTotalM  = stats.Int64(totalViolationsName, "Total number of violations per constraint", stats.UnitNone)
+	constraintsTotalM = stats.Int64(constraintsTotalName, "Total number of enforced constraints", stats.UnitNone)
+	auditLatencyM     = stats.Float64(auditLatency, "Latency of audit operation", stats.UnitMilliseconds)
+
+	methodTypeKey     = tag.MustNewKey("method_type")
+	constraintKindKey = tag.MustNewKey("constraint_kind")
+	constraintNameKey = tag.MustNewKey("constraint_name")
+)
+
+func init() {
+	register()
+}
+
+func register() {
+	views := []*view.View{
+		{
+			Name:        totalViolationsName,
+			Measure:     violationsTotalM,
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{methodTypeKey, constraintKindKey, constraintNameKey},
+		},
+		{
+			Name:        constraintsTotalName,
+			Measure:     constraintsTotalM,
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{methodTypeKey, constraintKindKey},
+		},
+		{
+			Name:        auditLatency,
+			Measure:     auditLatencyM,
+			Aggregation: view.Distribution(1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 11000, 12000, 13000, 14000, 15000),
```

Comment:
How did you choose these initial bucket values?

Reply:
Just chosen to get things started. Do you have any suggestions on how we can find optimal values?

Reply:
Definitely understand that. Note that the unit for these is now in seconds rather than milli or nano as they were before, so these need to be scaled down anyways. If you don't have quantitative samples yet, my suggestion is to start with a wide spectrum, eg: .005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10. These values give you more coverage than a linear spread which will help you find hot spots faster. Once you see which buckets are under or over populated, then you can adjust for a better distribution.

Reply:
updated with new values to match seconds